gdal/frmts/netcdf/netcdflayersg.cpp

```diff
@@ -0,0 +1,194 @@
+/* NetCDF Simple Geometry specific implementations
+ * for layers filled from CF 1.8 simple geometry variables
+ *
+ */
+#include "netcdfsg.h"
+#include "netcdfdataset.h"
+#include "ogr_core.h"
+
+namespace nccfdriver
+{
+	static OGRwkbGeometryType RawToOGR(geom_t type)
+	{
+		OGRwkbGeometryType ret = wkbNone;
+
+		switch(type)
+		{
+			case NONE:
+				break;
+			case LINE:
+				ret = wkbLineString;
+				break;
+			case MULTILINE:
+				ret = wkbMultiLineString;
+				break;
+			case POLYGON:
+				ret = wkbPolygon;
+				break;
+			case MULTIPOLYGON:
+				ret = wkbMultiPolygon;
+				break;
+			case POINT:
+				ret = wkbPoint;
+				break;
+			case MULTIPOINT:
+				ret = wkbMultiPoint;
+				break;
+			case UNSUPPORTED:
+				break;
+		}
+
+		return ret;	
+	}	
+
+}
+
+CPLErr netCDFDataset::DetectAndFillSGLayers(int ncid)
+{
+	// Discover simple geometry variables
+	int var_count;
+	nc_inq_nvars(ncid, &var_count);	
+	std::vector<int> vidList;
+
+	nccfdriver::scanForGeometryContainers(ncid, vidList);	
+
+	// To still do: support multiple geometry containers
+
+	for(size_t itr = 0; itr < vidList.size(); itr++)
+	{
+		try
+		{
+			LoadSGVarIntoLayer(ncid, vidList[itr]);
+
+		}
+		
+		catch(nccfdriver::SG_Exception * e)
```

Comment:
sure, we can do that :+1:   will fix

Reply:
Actually can I  do by pointer? I remembered that I decided to use pointer to take advantage of polymorphism of different exception objects.

Reply:
I also have a universal catch clause at the bottom that deletes the exception (the other one just rethrowing to that one) so the pathing is not that complex :)

Reply:
Polymorphism just works fine with exception thrown by value
The following displays yes!
```
struct A {};
struct B: public A {};
#include <stdio.h>
int main()
{
    try
    {
        throw B();
    }
    catch( const A& )
    {
        printf("yes!\n");
    }
    return 0;
}
```

Reply:
Well sure, sure... this I already knew.

But my implementation of SGeometry_Exception is fully virtual / abstact so I'm getting complaints about how SGeometry_Exception is not instantiable.

Reply:
If you catch by const ref, there's no reason it doesn't work. Below code just works:
```
struct A { virtual void foo() = 0; };
struct B: public A { void foo() override {} };
#include <stdio.h>
int main()
{
    try
    {
        throw B();
    }
    catch( const A& )
    {
        printf("yes!\n");
    }
    return 0;
}
```

Reply:
Hmm, maybe I forgot the const, thanks @rouault, I'll try again.

`offtopic: I also just like using pointers way more, in general since I do a decent amount of C`

Reply:
Strangely enough I tried it again and this time just using reference (also not needed const reference) works! Didn't know why it didn't work the first time...